diesel_derives/src/associations.rs

```diff
@@ -2,55 +2,84 @@ use syn;
 use quote;
 
 use model::{infer_association_name, Model};
-use util::str_value_of_meta_item;
+use util::{str_value_of_meta_item, wrap_item_in_const};
 
 pub fn derive_associations(input: syn::DeriveInput) -> quote::Tokens {
-    let mut derived_associations = Vec::new();
     let model = t!(Model::from_item(&input, "Associations"));
 
-    for attr in input.attrs.as_slice() {
+    let derived_associations = input.attrs.as_slice().iter().filter_map(|attr| {
         if attr.name() == "belongs_to" {
-            let options = t!(build_association_options(attr, "belongs_to"));
-            derived_associations.push(expand_belongs_to(&model, options))
+            Some(
+                build_association_options(attr)
+                    .map(|options| expand_belongs_to(&model, options))
+                    .unwrap_or_else(|e| e),
+            )
+        } else {
+            None
         }
-    }
+    });
 
-    quote!(#(#derived_associations)*)
+    wrap_item_in_const(
+        model.dummy_const_name("ASSOCIATIONS"),
+        quote!(#(#derived_associations)*),
+    )
 }
 
 fn expand_belongs_to(model: &Model, options: AssociationOptions) -> quote::Tokens {
     let parent_struct = options.name;
     let struct_name = &model.name;
+    let fields = model.attrs.as_slice();
 
     let foreign_key_name = options
         .foreign_key_name
         .unwrap_or_else(|| to_foreign_key(parent_struct.as_ref()));
+    let foreign_key_name = &foreign_key_name;
     let child_table_name = model.table_name();
-    let fields = model.attrs.as_slice();
+    let child_table_name = &child_table_name;
+    let foreign_key_attr = fields
+        .iter()
+        .find(|attr| attr.column_name().as_ref() == foreign_key_name.as_ref())
+        .unwrap();
```

Comment:
Can we use `expect` here? This is going to give a very poor error message if the model doesn't have a field with the right column name.